test/mpi/util/mtest.c

```diff
@@ -1022,6 +1022,33 @@ void MTestFreeComm(MPI_Comm * comm)
     }
 }
 
+/* Directly calling MTestGetIntercomm maybe insufficient since all the processes
+ * may end up with the same context_id even between different groups of the intercomm.
```

Comment:
We should still make sure to test the case where intercomm groups have the same context_id. Will this change prevent that?

Reply:
The  randomization is called from the tests explicitly. Currently only `pt2pt/icsend` and `pt2pt/icprobe` do that randomization. I guess we can run the tests twice to cover both cases.

Reply:
New commit added.